mlflow/pyfunc/cli.py

```diff
@@ -20,13 +61,22 @@ def commands():
 @cli_args.MODEL_PATH
 @cli_args.RUN_ID
 @click.option("--port", "-p", default=5000, help="Server port. [default: 5000]")
-def serve(model_path, run_id, port):
+@click.option("--no-conda", is_flag=True,
+              help="If specified, will assume that MLModel is running within a Conda environment "
+                   "with the necessary dependencies for the current project instead of attempting "
+                   "to create a new conda environment. Only valid if running locally.")
```

Comment:
Why "only valid if running locally"? I think if we move the `_rerun_in_conda` part to below the `if run_id`, and unconditionally pass `run_id=None` (because the new `model_path` is always local), then it should work.

(Tried it out locally; you should also be able to repro this by launching `mlflow ui` and setting `MLFLOW_TRACKING_URI=http://localhost:5000`. Although in my case I was using an S3 artifactory to uncover the run_id thing.)